stdlib/private/OSLog/OSLogMessage.swift

```diff
@@ -0,0 +1,285 @@
+/// when you pass a string interpolation to the log APIs.
+public struct OSLogInterpolation : StringInterpolationProtocol {
+  /// An OS log format string constructed from the given string interpolation.
+  public var formatString: String
+
+  /// A representation of a sequence of arguments that must be serialized
+  /// to bytes and passed to the OS logging system. The arguments are
+  /// (autoclosures of) expressions that are interpolated along with some
+  /// headers.
+  public var argumentSequence: OSLogArgumentSequence
+
+  /// The summary byte, as defined by the OS logging system, that summarizes the
+  /// nature/type of the arguments.
+  public var preamble: UInt8
+
+  /// Number of arguments i.e, number of interpolated expressions.
+  /// This will be determined on the fly in order to support concatenation
+  /// and interpolation of instances of `OSLogMessage`.
+  public var argumentCount: UInt8
+
+  public init(literalCapacity: Int, interpolationCount: Int) {
+    // TODO: format string must be fully constructed at compile time.
+    // The parameters `literalCapacity` and `interpolationCount` are ignored.
+    formatString = ""
+    argumentSequence = OSLogArgumentSequence()
+    preamble = 0
+    argumentCount = 0
+  }
+
+  public mutating func appendLiteral(_ literal: String) {
+    formatString += literal.percentEscapedString
+  }
+
+  /// Define interpolation for expressions of type Int. This definition enables
+  /// passing a formatting option and a privacy qualifier along with the
+  /// interpolated expression as shown below:
+  ///
+  ///         "\(x, as: .hex, is: .private\)"
+  ///
+  /// - Parameters:
+  ///  - number: the interpolated expression of type Int, which is autoclosured.
+  ///  - format: a formatting option available for Int types, defined by the
+  ///    enum `IntFormat`.
+  ///  - privacy: a privacy qualifier which is either private or public.
+  ///    The default is public.
+  public mutating func appendInterpolation(
+    _ number: @autoclosure @escaping () -> Int,
+    as format: IntFormat = .decimal,
+    is privacy: Privacy = .public) {
+    guard argumentCount < maxOSLogArgumentCount else { return }
+
+    addIntHeadersAndFormatSpecifier(
+      format,
+      isPrivate: privacy == .private,
+      byteSize: UInt8(MemoryLayout<Int>.size),
+      isUnsigned: false)
+    argumentSequence.append(number)
+  }
+
+  /// Construct/update format string and headers from the qualifiers (of the
+  /// interpolated expression) passed as parameters.
+  ///
+  /// All arguments to this function must be known at compile time.
+  public mutating func addIntHeadersAndFormatSpecifier(
+    _ format: IntFormat,
+    isPrivate: Bool,
+    byteSize: UInt8,
+    isUnsigned: Bool) {
+    // Always explicitly set either public or private flag bit.
+    let flag: UInt8 = isPrivate ? 1 : 2
+    formatString += getIntegerFormatSpecifier(
+      format,
+      isPrivate: isPrivate,
+      isLong: byteSize > 4,
```

Comment:
Just wondering, is it always 4? (I'm looking at you, arm64_32.)

Reply:
That's a good point.  I am thinking of changing the construction of this format specifier to be based on CTypes.swift, so that these constants are not hard coded. 
E.g. 

```swift
    switch byteSize {
    case _ where byteSize == UInt8(MemoryLayout<CInt>.size):
      break;
    case _ where byteSize == UInt8(MemoryLayout<CLongLong>.size):
      formatSpecifier += "ll"
    default:
      fatalError("Int size not supported. use integers with fixed size.")
    }
```